---
title: "Limma Tutorial"
author: Lindsay Rutter
output:
  packagedocs::package_docs:
  toc: true
toc_collapse: true
vignette: >
  %\VignetteEngine{packagedocs::redirect}
---
  
  <meta http-equiv="content-type" content="text/html;charset=utf-8" />
  
  ```{r global_options, include=FALSE}
# R output pre blocks are styled by default to indicate output
knitr::opts_chunk$set(
  comment = NA,
  cache = TRUE,
  fig.height = 8,
  fig.width = 10
)
```
Try new paper (https://www.bioconductor.org/help/workflows/RNAseq123/)

```{r}
library(limma)
library(Glimma)
library(GGally)
library(ggplot2)
library(edgeR)

thisPath <- getwd()
beeCounts <- readRDS("../data/data.Rds")
beeCounts <- as.matrix(beeCounts)
x <- DGEList(counts=beeCounts)

group <- factor(c(rep("NC",6), rep("NR",6), rep("VC",6), rep("VR",6)))
x$samples$group <- group
lane <- as.factor(c("L12","L12","L12","L12","L34","L34","L12","L12","L34","L34","L34","L34","L12","L12","L12","L12","L34","L34","L12","L12","L34","L34","L34","L34"))
x$samples$lane <- lane
```

Transform and remove low counts.
```{r}
cpm <- cpm(x)
keep.exprs <- rowSums(cpm>1)>=4 # tried filtering up to 24 and not much difference
x <- x[keep.exprs,, keep.lib.sizes=FALSE] # 15,314 to 10,654
dim(x)
x <- calcNormFactors(x, method = "TMM")
```

Make boxplots
```{r}
ggparcoord(data.frame(log(x[[1]]/colMeans(x[[1]]))), columns=1:24, alphaLines=0, boxplot=TRUE, scale="globalminmax") + coord_flip()
```

Create MDS plots
```{r}
library(RColorBrewer)
lcpm <- cpm(x, log=TRUE)
par(mfrow=c(1,2))
col.group <- group
levels(col.group) <-  brewer.pal(nlevels(col.group), "Set1")
col.group <- as.character(col.group)
col.lane <- lane
levels(col.lane) <-  brewer.pal(nlevels(col.lane), "Set2")
col.lane <- as.character(col.lane)
```

```{r}
plotMDS(lcpm, labels=colnames(lcpm), col=col.group)
```

```{r}
plotMDS(lcpm, labels=colnames(lcpm), col=col.lane, dim=c(3,4))
```

```{r}
glMDSPlot(lcpm, labels=paste(group, lane, colnames(lcpm), sep="_"), groups=x$samples[,c(1,4)], launch=FALSE)
```

Create design matrix. There are many ways to setup a design matrix. Here, we removed the intercept from group (the first factor), but kept the intercept from lane. This allows us to do contrasts with group more easily.
```{r}
design <- model.matrix(~0+group+lane)
colnames(design) <- gsub("group", "", colnames(design))

contr.matrix <- makeContrasts(
   NC_NR = NC-NR,
   NC_VC = NC-VC,
   NC_VR = NC-VR,
   NR_VC = NR-VC,
   NR_VR = NR-VR,
   VC_VR = VC-VR,
   levels = colnames(design))
```

```{r}
par(mfrow=c(1,2))
v <- voom(x, design, plot=TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts=contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main="Final model: Mean−variance trend")
```

For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted p-value cutoff that is set at 5% by default. For the comparison between expression levels in basal and LP, 4,127 genes are found to be down-regulated in basal relative to LP and 4,298 genes are up-regulated in basal relative to LP – a total of 8,425 DE genes.

```{r}
summary(decideTests(efit))
```

```{r}
pairNames <- colnames(contr.matrix)
metrics <- list()
for (i in 1:length(pairNames)) {
  temp <- topTreat(efit, coef=i, n=Inf)
  temp <- cbind(ID=rownames(temp), temp)
  temp$ID = as.character(temp$ID)
  metrics[[ pairNames[i] ]] <- temp
}
```

Save the data

```{r}
saveRDS(metrics, file="dataMetrics.Rds")
saveRDS(x, file="data.Rds")
```
